fenix/app/src/main/java/org/mozilla/fenix/debugsettings/ui/DebugOverlay.kt

```diff
@@ -41,12 +67,38 @@ fun DebugOverlay() {
                 .align(Alignment.CenterStart)
                 .padding(start = 16.dp),
             onClick = {
-                scope.launch {
-                    snackbarState.showSnackbar("Show debug drawer")
-                }
+                drawerEnabled = true
             },
         )
 
+        // ModalDrawer utilizes a Surface, which blocks ALL clicks behind it, preventing the app
+        // from being interactable. This cannot be overridden in the Surface API, so we must hide
+        // the entire drawer when it is closed. Whenever the app is Compose from
+        // the root layer, we can remove this.
+        if (drawerEnabled) {
+            val currentLayoutDirection = LocalLayoutDirection.current
+            val sheetLayoutDirection = when(currentLayoutDirection) {
+                LayoutDirection.Rtl -> LayoutDirection.Ltr
+                LayoutDirection.Ltr -> LayoutDirection.Rtl
+            }
+
+            // Force the drawer to always open from the opposite side of the screen
+            CompositionLocalProvider(LocalLayoutDirection provides sheetLayoutDirection) {
+                ModalDrawer(
+                    drawerContent = {
+                        // Ensure the drawer's content still respects the device's direction setting
+                        CompositionLocalProvider(LocalLayoutDirection provides currentLayoutDirection) {
+                            DebugDrawer()
+                        }
+                    },
+                    drawerBackgroundColor = FirefoxTheme.colors.layer1,
+                    scrimColor = FirefoxTheme.colors.scrim,
+                    drawerState = drawerState,
+                    content = {},
+                )
+            }
+        }
+
         // This must be the last element in the Box
         SnackbarHost(
```

Comment:
Do we still need this?

Reply:
Should we make a generic component?

Reply:
Potentially! We already have a [ticket](https://bugzilla.mozilla.org/show_bug.cgi?id=1814198) for it, actually. I think GL started on it a long time ago, and probably just needs to be taken over and restarted.

Reply:
There's a chance that the Snackbar is on the official roadmap for 2024, if the grid design system project gets prioritized, but we can still try to work on this sooner than that.